lib/iris/coords.py

```diff
@@ -1391,6 +1394,9 @@ def __eq__(self, other):
 
     # The __ne__ operator from Coord implements the not __eq__ method.
 
+    def __hash__(self):
+        return hash(id(self))
```

Comment:
You're right it's probably inconsistent, but this is the simplest hash and it should be equivalent to what occurs on Python 2.